### PR TITLE
Capture panics

### PR DESCRIPTION
**Pull request by skoushan**
Captures panics in addition to jobs that return an error.

### CODE REVIEW

File: sentry.go [hunk 1]
```diff
@@ -1,17 +1,30 @@
...
-		if err != nil {
+
+		if err != nil { // capture errors returned by job
 			packet := raven.NewPacket(err.Error(), raven.NewException(err, raven.NewStacktrace(1, 3, nil)))
 			packet.Extra["Job"] = job
 			packet.Extra["Arguments"] = args
 			raven.Capture(packet, map[string]string{"job": job, "logger": "resque"})
 		}
+
+		if e := recover(); e != nil { // capture panics
```
Comment (cmelbye) [inline sentry.go]: `recover()` needs to be in a defer that's set-up before the code that you want to recover from. Like this:

```go
defer func() {
		if e := recover(); e != nil { // capture panics
			eStr := fmt.Sprint(e)
			packet := raven.NewPacket(eStr, raven.NewException(errors.New(eStr), raven.NewStacktrace(1, 3, nil)))
			raven.Capture(packet, map[string]string{"job": job, "logger": "resque"})
		}
}()

err := w(queue, args...)
// etc
```

Comment (skoushan) [inline sentry.go]: ooops you're right thanks!

File: sentry.go [hunk 2]
```diff
@@ -1,17 +1,32 @@
...
+	"github.com/getsentry/raven-go"
+)
 
 // Wrapper wraps goworkers and reports failed jobs to Sentry
 func Wrapper(job string, w func(string, ...interface{}) error) func(string, ...interface{}) error {
 	return func(queue string, args ...interface{}) error {
+		defer func() {
+			if e := recover(); e != nil { // capture panics
+				eStr := fmt.Sprint(e)
+				packet := raven.NewPacket(eStr, raven.NewException(errors.New(eStr), raven.NewStacktrace(1, 3, nil)))
```
Comment (cmelbye) [inline sentry.go]: Are these the right values for the stacktrace?

Comment (skoushan) [inline sentry.go]: Yes, the `1` skips the `packet := raven.NewPacket(...)` line, and the 3 is the default value used in the package for "surrounding" context around a line

Comment (skoushan) [inline sentry.go]: I'm going to extract out those lines into a function

**Pull request merged**